package/src/main/java/com/azure/cosmos/cassandra/CosmosRetryPolicy.java

```diff
@@ -0,0 +1,301 @@
+
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.config.DriverOption;
+import com.datastax.oss.driver.api.core.connection.ClosedConnectionException;
+import com.datastax.oss.driver.api.core.connection.HeartbeatException;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.api.core.retry.RetryDecision;
+import com.datastax.oss.driver.api.core.retry.RetryPolicy;
+import com.datastax.oss.driver.api.core.servererrors.CoordinatorException;
+import com.datastax.oss.driver.api.core.servererrors.OverloadedException;
+import com.datastax.oss.driver.api.core.servererrors.WriteFailureException;
+import com.datastax.oss.driver.api.core.servererrors.WriteType;
+import com.datastax.oss.driver.api.core.session.Request;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import java.util.Objects;
+import java.util.Random;
+import java.util.function.BiFunction;
+
+/**
+ * A {@link RetryPolicy} implementation with back-offs for {@link CoordinatorException} failures.
+ * <p>
+ * This is the default retry policy when you take a dependency on this package. It provides a good out-of-box
+ * experience for communicating with Cosmos Cassandra API instances. Its behavior is specified in configuration:
+ * <pre>{@code
+ * datastax-java-driver.advanced.retry-policy {
+ *   class = DefaultRetryPolicy
+ *   max-retries = 5              # Maximum number of retries
+ *   fixed-backoff-time = 5000    # Fixed backoff time in milliseconds
+ *   growing-backoff-time = 1000  # Growing backoff time in milliseconds
+ * }
+ * }</pre>
+ * The number of retries that should be attempted is specified by {@code max-retries}. A value of {@code -1} specifies
+ * that an indefinite number of retries should be attempted. For {@link #onReadTimeout onReadTimout},
+ * {@link #onWriteTimeout onWriteTimeout}, and {@link #onUnavailable onUnavailable}, retries are immediate. For
+ * {@link #onErrorResponse onResponse}, the exception message is parsed to obtain the value of the {@code RetryAfterMs}
+ * field provided by the server as the back-off duration. If {@code RetryAfterMs} is not available, the default
+ * exponential growing back-off scheme is used. In this case the time between retries is increased by
+ * {@code growing-backoff-time} on each retry, unless {@code max-retries} is {@code -1}. In this case back-off occurs
+ * with fixed {@code fixed-backoff-time} duration.
+ *
+ * @see <a href="../../../../../doc-files/reference.conf.html">reference.conf</a>
+ * @see <a href="https://docs.datastax.com/en/developer/java-driver/latest/manual/core/retries/">DataStax Java
+ * Driver Retries</a>
+ */
+public final class CosmosRetryPolicy implements RetryPolicy {
+
+    // region Fields
+
+    private static final int GROWING_BACKOFF_SALT_IN_MILLIS = 2_000;
```

Comment:
is 2_000 2000 or 2 Ms? Was this the default from the v3 policy?

Reply:
I picked this value up from the v3 driver where the value is 2 seconds. Were you thinking it should be 2 ms?